crates/torii/core/src/engine.rs

```diff
@@ -303,26 +308,26 @@ impl<P: Provider + Sync> Engine<P> {
                     }
                 }
 
-                if let Some((_, last_tx_hash)) = transactions.last() {
-                    // Dedup transactions
-                    // As me might have multiple events for the same transaction
-                    if *last_tx_hash == event.transaction_hash {
-                        continue;
-                    }
-                }
-                transactions.push((block_number, event.transaction_hash));
+                transactions
+                    .entry((block_number, event.transaction_hash))
+                    .or_insert(vec![])
+                    .push(event);
             }
         }
 
+        debug!("Transactions: {}", &transactions.len());
+        debug!("Blocks: {}", &blocks.len());
         // Process all transactions
         let mut last_block = 0;
-        for (block_number, transaction_hash) in transactions {
+        for ((block_number, transaction_hash), events) in transactions {
+            debug!("Processing transaction hash: {:#x}", transaction_hash);
             // Process transaction
             let transaction = self.provider.get_transaction_by_hash(transaction_hash).await?;
 
-            self.process_transaction_and_receipt(
+            self.process_transaction_with_events(
```

Comment:
What is the probability that we didn't fetch all the events of a transaction? I think it's null, but wanted to confirm. Because the `EventFilter` is always block based. So theoretically, even if we have small chunks, as we parse all the chunks, we should always end up with a list of all the transactions for a given block + all transaction's events.

Reply:
(discussed on a call)